### DISCUSSION TITLE
best metric, accuracy = 0.0

### DISCUSSION THREAD

**Discussion by Balicki24** [PaddlePaddle/PaddleOCR#14054] [Q&A]
I'm finetuning the Recognition Model for Japanese Language . But when training is complete , i see :
ppocr INFO: best metric, acc: 0.0, norm_edit_dis: 0.7167110076645224, fps: 2079.1379306804797, best_epoch: 53, start_epoch: 371, is_float16: False

But when training, the best acc = 0.95 >
I dont know what make wrong

**Comment by GreatV** [comment]
Not enough information to determine the cause.

**Reply by Balicki24** [reply]
What information you need to find the cause, i will give you.
 Pls help me

**Reply by Balicki24** [reply]
The lastest training epoch, the acc = 0.95 but the model save the state when acc =0

**Comment by GreatV** [comment]
Running environment, configuration file, minimal reproducible demo, dataset size.

**Reply by Balicki24** [reply]
This is my config file :
![image](https://github.com/user-attachments/assets/91897a81-20a8-4962-a337-225bc3ca1882)
![image](https://github.com/user-attachments/assets/940039f8-5acb-43ed-91af-7e54d8a4a356)
![image](https://github.com/user-attachments/assets/3b614bfd-28d5-4167-bdc1-a6b8464989f9)
I running in 2 GPU T4 in kaggle with the train dataset and val dataset is the same to overfit the model. The size of dataset is about 1036 image i cutting from some japanese company bills . 
Some of image like that : 
![image1_1_4](https://github.com/user-attachments/assets/07dacd84-be5d-4609-b98f-fa3d52fc1b54)
![image1_1_12](https://github.com/user-attachments/assets/1f562062-ade5-4e2e-8d3a-1be140a802bd)

And the lastest epoch is : 
![image](https://github.com/user-attachmme

**Reply by GreatV** [reply]
Can you provide more training logs?

**Reply by Balicki24** [reply]
Here are some information in my trainning logs: 
![image](https://github.com/user-attachments/assets/44246da8-076e-41ad-a228-e44fc0e0a9ce)
![image](https://github.com/user-attachments/assets/058d9c44-d48e-4742-914d-c3b2bce55cf9)
![image](https://github.com/user-attachments/assets/687645f8-5da9-4ea1-87c7-eb5fa3e24c7b)

![image](https://github.com/user-attachments/assets/1cb2b8a9-754d-4aeb-a62e-a2cea1a81107)
![image](https://github.com/user-attachments/assets/20c0d2c1-4312-4fbd-a73f-7da4eacdfdfd)

**Reply by GreatV** [reply]
Based on your provided details, it seems like you're experiencing an issue where the accuracy reported during training is quite high (up to 0.95), but after training completion, the best metric accuracy is reported as 0.0. 

Here are some potential reasons and debugging steps:

### 1. **Validation Dataset Setup (Overfitting Check)**
   - You mentioned that both the train and validation datasets are the same (1036 images from company bills). Since you're overfitting, the accuracy should theoretically be close to 1.0, both during training and validation. However, if the evaluation stage is reporting an accuracy of 0.0, this could indicate:
     - The validation data is not being loaded or processed correctly during evaluation.
     - Some discrepancy in how validation labels are matched compared to training labels.
     - Overfitting shouldn't cause this, but incorrect setup of datasets for validation can.

### 2. **Check Dataset Path in Configuration**
   - The error messages in the logs show multiple instances of:
     ```python
     Exception: "image_path does not exist"
     ```
     This suggests there is a path error when the model tries to load images for validation (or even during training). Make sure that all paths to images are correct and accessible, especially in your validation set. If validation data is missing or paths are incorrect, this could result in the accuracy being 0 at the end of training.

### 3. **Metrics Computation During Validation**
   - The metric `norm_edit_dis` seems to be computed correctly (values like 0.7167 during the evaluation stage). However, the accuracy computation might be breaking. This could be due to:
     - Incorrect label mapping during validation.
     - Issues in how accuracy is calculated at the end compared to during training.

**Reply by Balicki24** [reply]
Thank you for your suport . I will find the problem in my config file . The eval_batch_step was too large so that the model can't save the best metric in last epoch
